src/studio/src/designer/frontend/ux-editor/features/formDesigner/formLayout/formLayoutSagas.ts

```diff
@@ -458,18 +458,22 @@ export function* watchUpdateLayoutNameSaga(): SagaIterator {
 export function* fetchFormLayoutSettingSaga(): SagaIterator {
   try {
     const settings: ILayoutSettings = yield call(get, getLayoutSettingsUrl());
-    yield put(FormLayoutActions.fetchLayoutSettingsFulfilled({ settings }));
+    if (settings) {
+      yield put(FormLayoutActions.fetchLayoutSettingsFulfilled({ settings }));
+    }
   } catch (error) {
     yield put(FormLayoutActions.fetchLayoutSettingsRejected({ error }));
   }
 }
 
 export function* watchFetchFormLayoutSettingSaga(): SagaIterator {
-  yield all([
-    take(FormLayoutActions.fetchFormLayoutFulfilled),
-    take(FormLayoutActions.fetchLayoutSettings),
-  ]);
-  yield call(fetchFormLayoutSettingSaga);
+  while (true) {
```

Comment:
I'm not completely sure why there is a while here,
but are there any cases where the watchFetchFromLayoutSettingsSaga will be initiated more than once?
If so, then the loop needs some way to break, or you can have an overflow on your hands.

Reply:
While there to handle fetching layoutsetting when user navigates back to ux-designer. The way this was implemented now the layout-settings would not get fetched (but formlayout would, which would override layout order). This makes sure layoutsettings is fetched in the same way as formlayout.

Generator functions are special, they can be thought of like pausable functions. Even the `redux-saga` official effects are written using while loops that never break. Discussion about this if interested: https://github.com/redux-saga/redux-saga/issues/1116

Reply:
I guess this could be rewritten to
```
    yield takeEvery([
      take(FormLayoutActions.fetchFormLayoutFulfilled),
      take(FormLayoutActions.fetchLayoutSettings),
    ]);
```
To make it a little more readable.

Reply:
From what I understand is that generators primarily work as co-routines.
With a while(true) the generator will never go out of memory. So if for some reason it would be run several times, it could build up to several instances of the same generator in memory.
I'm new to saga, but I assume that saga is handling the next() calls on the generators, and has some sort of garbage collection once a generator is no longer needed.
With a while(true) surrounding when does the generator actually end?

Reply:
@mjulstein jupp, saga handles calling next for us. For this particular case, we don't actually want the watcher saga to actually end, we want this to listen for these two actions "forever". The garbage collection part I'm not actually sure about, but please share your knowledge if you go investigating this. 

But good catch to refactor this to be more readable. Fix coming :)